Typecheck errors with latest Nuxt v4.3.0 update

### Environment

**Operating system** | `Linux 5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/144.0.0.0 Safari/537.36` |
| **CPU**              | `Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz (4 cores)`                                                  |
| **Node.js version**  | `v20.19.1`                                                                                            |
| **nuxt/cli version** | `3.32.0`                                                                                              |
| **Package manager**  | `pnpm@8.15.6`                                                                                         |
| **Nuxt version**     | `4.3.0`                                                                                               |
| **Nitro version**    | `2.13.1`                                                                                              |
| **Builder**          | `vite@7.3.1`                                                                                          |
| **Config**           | `compatibilityDate`, `devtools`, `modules`, `runtimeConfig`                                           |
| **Modules**          | `@nuxtjs/i18n@10.2.1`                                                                                 |

### Reproduction

https://stackblitz.com/edit/github-fhycavek

### Describe the bug

This module was working well with the previous version of Nuxt (v4.2.2).
By updating to Nuxt 4.3.0, typecheck now reports errors about the imports in composables.
By disabling this module, no typecheck errors are happening.

### Additional context

In the provided reproduction, launch `pnpm typecheck`.
See the errors.

### Logs

```shell
❯ pnpm typecheck

> nuxt-app@ typecheck /home/projects/github-fhycavek
> nuxt typecheck


[8:55:36 AM]  ERROR  (node:75) ExperimentalWarning: WASI is an experimental feature and might change at any time
(Use node --trace-warnings ... to show where the warning was created)


[nuxt-i18n 8:55:39 AM]  WARN  The prefix_except_default i18n strategy (used by default) needs defaultLocale to be set.

app/composables/useTest.ts:3:19 - error TS2304: Cannot find name 'computed'.

3   const appName = computed<string>(() => config.public.app.name);
                    ~~~~~~~~

app/composables/useTest.ts:5:16 - error TS2304: Cannot find name 'ref'.

5   const test = ref<number>(1);
                 ~~~

app/composables/useTest.ts:6:17 - error TS2304: Cannot find name 'computed'.

6   const testC = computed<number>(() => test.value);
                  ~~~~~~~~


Found 3 errors.


 ERROR  Process exited with non-zero status (2)
```

--- a/src/runtime/utils.ts
+++ b/src/runtime/utils.ts
@@ -43,8 +43,8 @@ export type ComposableContext = {
     /** Enable/disable hreflangLinks */
     hreflangLinks: boolean
   }
-  head: ReturnType<typeof useHead>
-  _head: ReturnType<typeof useHead> | undefined
+  head: ReturnType<typeof import('nuxt/app').useHead>
+  _head: ReturnType<typeof import('nuxt/app').useHead> | undefined
   metaState: Required<I18nHeadMetaInfo>
   seoSettings: I18nHeadOptions
   localePathPayload: Record<string, Record<string, string> | false>

